File: nodejs/cucumber/pages/ajax.page.ts
import { expect, type Locator, type Page } from '@playwright/test';

export class AjaxPage {

  page: Page;
  button: Locator;
  title: Locator;

  constructor(page: Page) {
    this.page = page;
    this.button = page.locator("#button");
    this.title = page.locator("#title");
  }

  async click() {
    await this.button.click();
  }

  async wait_event() {
    await expect(this.title).toBeVisible({ timeout: 15000 });
  }

  async verify_title() {
    await expect(this.title).toHaveText("AJAX");
  }

}

File: python/pytest/web_playwright/pages/ajax_page.py
from playwright.sync_api import Page, expect


class AjaxPage:

    def __init__(self, page: Page):
        self.page = page
        self.button = page.locator("#button")
        self.title = page.locator("#title")

    def click(self):
        self.button.click()

    def wait_event(self):
        expect(self.title).to_be_visible(timeout=15_000)

    def verify_title(self):
        expect(self.title).to_have_text("AJAX")
